src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java

```diff
@@ -346,10 +340,7 @@ public Txn createTxn(ClientState state, QueryOptions options)
             List<TxnNamedRead> reads = createNamedReads(options, state, null, keySet::add);
             Keys txnKeys = toKeys(keySet);
             TxnKeyRead read = createTxnRead(reads, null);
-            Txn.Kind kind = txnKeys.size() == 1
-                    && transactionalModeForSingleKey(txnKeys) == TransactionalMode.full
-                    && DatabaseDescriptor.getAccordEphemeralReadEnabledEnabled()
-                    ? EphemeralRead : Read;
+            Txn.Kind kind = shouldReadEphemerally(txnKeys, Schema.instance.getTableMetadata(((AccordRoutableKey) txnKeys.get(0)).table()).params, Read);
```

Comment:
In the process of setting up and routing an operation we fetch the table parameters repeatedly from the cluster metadata. In a lot of cases we have already fetched them from the `ClusterMetadata` and could re-use them. I don't really have a preference since in the big picture this isn't the thing that is making C* slow.